src/configurePipeline/clients/azureDevOpsClient.ts

```diff
@@ -0,0 +1,241 @@
+                "type": "github",
+                "url": "http://github.com"
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        });
+    }
+
+    public async createAzureServiceConnection(endpointName: string, inputs: WizardInputs, scope?: string, ): Promise<any> {
+        return this.serviceClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: "https://" + inputs.organizationName + ".visualstudio.com/" + inputs.projectName + "/_apis/serviceendpoint/endpoints",
+            headers: {
+                "Content-Type": "application/json",
+                "Accept": "application/json;api-version=5.1-preview.2;excludeUrls=true"
+            },
+            method: "POST",
+            body: {
+                "administratorsGroup": null,
+                "authorization": {
+                    "parameters": {
+                        "authenticationType": "spnKey",
+                        "scope": scope,
+                        "serviceprincipalid": "",
+                        "serviceprincipalkey": "",
+                        "tenantid": inputs.authDetails.tenantId
+                    },
+                    "scheme": "ServicePrincipal"
+                },
+                "data": {
+                    "appObjectId": "",
+                    "azureSpnPermissions": "",
+                    "azureSpnRoleAssignmentId": "",
+                    "creationMode": "Automatic",
+                    "environment": "AzureCloud",
+                    "scopeLevel": "Subscription",
+                    "spnObjectId": "",
+                    "subscriptionId": inputs.subscriptionId,
+                    "subscriptionName": inputs.subscriptionId
+                },
+                "description": "",
+                "groupScopeId": null,
+                "name": endpointName,
+                "operationStatus": null,
+                "readersGroup": null,
+                "type": "azurerm",
+                "url": "https://management.azure.com/"
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        });
+    }
+
+    public async getEndpointStatus(endpointId: string, organizationName: string, projectName: string, ): Promise<any> {
+        return this.serviceClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: "https://" + organizationName + ".visualstudio.com/" + projectName + "/_apis/serviceendpoint/endpoints/" + endpointId,
+            headers: {
+                "Content-Type": "application/json",
+                "Accept": "application/json;api-version=5.1-preview.2;excludeUrls=true"
+            },
+            method: "Get",
+            deserializationMapper: null,
+            serializationMapper: null
+        });
+    }
+
+    public async authorizeEndpointForAllPipelines(endpointId: string, endpointName: string, organizationName: string, projectName: string): Promise<any> {
```

Comment:
is it a two step process or we can do it while creating a new endpoint

Reply:
This is required to be done explicitly, after the endpoint is ready.